spec/controllers/meetings_controller_spec.rb

```diff
@@ -248,4 +248,134 @@
+        end
+        
+        it { expect(response).to render_template(:new) }
+      end
+
+      context 'when params are valid' do
+        context 'user is the group leader' do
+          it 'creates a new meeting' do
+            expect { post :create, params: { meeting: valid_meeting_params }}
+              .to change(Meeting, :count).by 1
+          end
+
+          it 'creates a new meeting_member' do
+            expect { post :create, params: { meeting: valid_meeting_params }}
+              .to change(MeetingMember, :count).by 1
+          end
+
+          it 'redirects to group path' do
+            post :create, params: { meeting: valid_meeting_params }
+            expect(response).to redirect_to group_path(group.id)
+          end
+        end
+
+        context 'user is not the leader' do
+          before do
+            group_member.update!(leader: false)
+            post :create, params: { meeting: valid_meeting_params }
+          end
+
+          it { expect(response).to redirect_to group_path(group.id) }
+        end
+      end
+    end
+  end
+
+  describe 'PATCH/PUT #update' do
+    let(:meeting) { create(:meeting) }
+    let!(:user) { create(:user) }
+    let(:valid_update_params) { attributes_for(:meeting).merge(name: 'updated name') }
+    let(:invalid_update_params) { { name: nil, description: nil } }
+
+    context 'when the user is logged in' do
+      include_context :logged_in_user
+      context 'when valid params are supplied' do
+        before { patch :update, params: { id: meeting.id, meeting: valid_update_params } }
+        it 'updates the meeting' do
+          expect(meeting.reload.name).to eq 'updated name'
+        end
+        it 'redirects to the meeting page' do
+          expect(response).to redirect_to meeting_path(meeting.reload.slug)
+        end
+      end
+      context 'when invalid params are supplied' do
+        before { patch :update, params: { id: meeting.id, meeting: invalid_update_params } }
+        it 're-renders the edit form' do
+          expect(response).to render_template(:edit)
+        end
+      end
+    end
+    context 'when the user is not logged in' do
+      before { patch :update, params: { id: meeting.id } }
+      it_behaves_like :with_no_logged_in_user
+    end
+  end
+
+  describe 'DELETE #destroy' do
+    let(:user) { create(:user) }
+    let(:group) { create(:group) }
+    let(:meeting) { create(:meeting, group_id: group.id) }
+    let!(:group_member) { create(:group_member, group_id: meeting.group_id, user: user, leader: true) }
+    let(:meeting_member) { create(:meeting_member, user: user, meeting: meeting) }
+
+    context 'when the user is logged in' do
+      include_context :logged_in_user
+      
+      context 'user is the group leader' do
+        it 'deletes the meeting members' do
+          expect { delete :destroy, params: { id: meeting.id } }
+            .to change(MeetingMember, :count).by(-1)
+        end
+
+        it 'deletes the meeting' do
```

Comment:
I have tried 

```ruby
it 'deletes the meeting' do
    expect(meeting).to receive(:destroy)
    delete :destroy, params: { id: meeting.id }
end
```
and 
```ruby
it 'deletes the meeting' do
    expect(meeting).to receive(:destroy)
    delete :destroy, params: { id: meeting.id }
end
``` 
but all fail with 
```shell
expected: 1 time with any arguments
received: 0 times with any arguments
```

Where I'm I going wrong? @jm96441n

Reply:
@jm96441n if you get time please respond to this. Thanks

Reply:
Let me pull down your branch and give it a try, that seems like it should be working but there might be a difference in the instance of the meeting being retrieved in the controller and the instance being stubbed here. This is an instance where mocks/stubs might make this a bit easier by using a mock in the setup (`build :meeting`) and stubbing the find method on the `Meeting` class you can ensure that you're working with the same object in the test and the controller. I'll post back later tonight when I have a chance to check this out though

Reply:
Okay, so the issue comes from the fact that `@meeting` in the controller and `meeting` in the spec refer to the same database row, but are two different instantiated ruby objects. The fix would be to stub out the find method on `Meeting` so that it returns the `meeting` object instantiated in the spec (example is the spec on line 356):
```
it 'deletes the meeting' do
  # Stub the 'set_meeting' method to return the object we instantiate in our spec
  allow(Meeting).to receive_message_chain(:friendly, :find).and_return(meeting)
  expect(Meeting).to receive(:destroy)
  delete :destroy, params: { id: meeting.id }
end
```

Reply:
Doing this also allows us to use 
```
let(:meeting) { build(:meeting) }
```
instead of create so we can remove the DB hits in our unit tests and get closer to isolated unit tests where we're testing only the logic we write

Reply:
Great! Thanks @jm96441n